server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -720,6 +722,8 @@ public void getLocalShuffleData(
           sdr.release();
         }
         shuffleServer.getShuffleBufferManager().releaseReadMemory(length);
+        ShuffleServerMetrics.gaugeReadLocalDataFileThreadNum.dec();
```

Comment:
This is not accurate to dec metric here, because the `sdr = shuffleServer.getShuffleTaskManager().getShuffleData` will throw exception before the `inc` happen.

Reply:
Done.